File: CS/ChangeDatabase.Module.Web/WebChangeDatabaseController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Web.SystemModule;
using System.Web;
using DevExpress.ExpressApp.Web;
using System.Web.Security;

namespace ChangeDatabase.Module.Web {
    public class WebChangeDatabaseController : WindowController {
        public const string DatabaseParameterName = "DatabaseName";
        public SingleChoiceAction changeDatabaseAction;
        public WebChangeDatabaseController() {
            this.TargetWindowType = WindowType.Main;

            changeDatabaseAction = new SingleChoiceAction(this, "ChangeDatabase", WebApplication.OldStyleLayout ? "Tools" : "Security");
            foreach(string databaseName in ChangeDatabaseHelper.Databases.Split(';')) {
                changeDatabaseAction.Items.Add(new ChoiceActionItem(databaseName, databaseName));
            }
            changeDatabaseAction.Execute += new SingleChoiceActionExecuteEventHandler(changeDatabaseAction_Execute);
        }

        protected override void OnActivated() {
            base.OnActivated();
            foreach(ChoiceActionItem item in changeDatabaseAction.Items) {
                if(Application.ConnectionString.Contains((string)item.Data)) {
                    changeDatabaseAction.SelectedItem = item;
                    break;
                }
            }
        }
        void changeDatabaseAction_Execute(object sender, SingleChoiceActionExecuteEventArgs e) {
            SecuritySystem.Instance.Logoff();
            HttpContext.Current.Session.Abandon();
            WebApplication.Redirect(FormsAuthentication.DefaultUrl + "?" + DatabaseParameterName + "=" + (string)e.SelectedChoiceActionItem.Data);
        }
    }
}

File: VB/ChangeDatabase.Module.Web/WebChangeDatabaseController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Web
Imports System.Web.Security
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Web
Imports DevExpress.ExpressApp.Actions

Namespace ChangeDatabase.Module.Web
	Public Class WebChangeDatabaseController
		Inherits WindowController
		Public Const DatabaseParameterName As String = "DatabaseName"
		Public changeDatabaseAction As SingleChoiceAction
		Public Sub New()
			Me.TargetWindowType = WindowType.Main

			changeDatabaseAction = New SingleChoiceAction(Me, "ChangeDatabase", "Security")
			For Each databaseName As String In ChangeDatabaseHelper.Databases.Split(";"c)
				changeDatabaseAction.Items.Add(New ChoiceActionItem(databaseName, databaseName))
			Next databaseName
			AddHandler changeDatabaseAction.Execute, AddressOf changeDatabaseAction_Execute
		End Sub

		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			For Each item As ChoiceActionItem In changeDatabaseAction.Items
				If Application.ConnectionString.Contains(CStr(item.Data)) Then
					changeDatabaseAction.SelectedItem = item
					Exit For
				End If
			Next item
		End Sub
		Private Sub changeDatabaseAction_Execute(ByVal sender As Object, ByVal e As SingleChoiceActionExecuteEventArgs)
			SecuritySystem.Instance.Logoff()
			HttpContext.Current.Session.Abandon()
			WebApplication.Redirect(FormsAuthentication.DefaultUrl & "?" & DatabaseParameterName & "=" & CStr(e.SelectedChoiceActionItem.Data))
		End Sub
	End Class
End Namespace
